SETUP.md

```diff
@@ -0,0 +1,184 @@
+# Setting up your Cook dev environment
+
+This document tells you how to set up a Cook dev environment from
+scratch on a Macintosh computer.
```

Comment:
I really like these docs I think they are very helpful. Would it be possible to separate out the mac specific instructions and notes from the general notes?